geoh5py/workspace/workspace.py

```diff
@@ -1228,22 +1208,29 @@ def open(self, mode: str | None = None) -> Workspace:
 
         return self
 
-    def _register_type(self, entity_type: EntityType):
-        weakref_utils.insert_once(self._types, entity_type.uid, entity_type)
-
-    def _register_group(self, group: Group):
-        weakref_utils.insert_once(self._groups, group.uid, group)
-
-    def _register_data(self, data_obj: Entity):
-        weakref_utils.insert_once(self._data, data_obj.uid, data_obj)
+    def register(self, entity: Entity | EntityType | PropertyGroup):
+        """
+        Register an entity to the workspace based on its type.
 
-    def _register_object(self, obj: ObjectBase):
-        weakref_utils.insert_once(self._objects, obj.uid, obj)
+        :param entity: The entity to be registered.
+        """
+        dict_entity: dict[uuid.UUID, ReferenceType]
+        if isinstance(entity, EntityType):
+            dict_entity = self._types
```

Comment:
Wowzer, nope. This is not doing anything right now. We still need the `weak_ref.insert_once` for every case.